lib/util/esWrapper.js

```diff
@@ -211,13 +210,81 @@ class ESWrapper {
         }
 
         if (data && data.index && Object.keys(picked).length === 0) {
-          throw new NotFoundError(`No mapping found for index "${data.index}"`);
+          errorsManager.throw(
+            'external',
+            'elasticsearch',
+            'no_mapping_found',
+            data.index);
         }
 
         return picked;
       })
       .catch(error => Bluebird.reject(this.formatESError(error)));
   }
+
+  _handleConflictError (error, message) {
+    debug('unhandled "Conflict" elasticsearch error: %a', error);
+
+    return errorsManager.getError(
+      'external',
+      'elasticsearch',
+      'unexpected_conflict_error',
+      message);
+  }
+
+  _handleNotFoundError (error, message) {
+    // Check if it's a 404 on a document
+    if (error.meta.body.found === false) {
+      return errorsManager.getError(
+        'external',
+        'elasticsearch',
+        'document_not_found',
+        error.meta.body._id);
+    }
+
+    if (error.meta.body && error.meta.body.error) {
+      message = error.meta.body.error
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Introduce a new variable instead of reusing the parameter "message". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1226)